File: 2015/go/day-01/day01.go
package main

import (
	"fmt"
	"os"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	res, err := os.ReadFile("2015/go/day-01/input.txt")

	check(err)

	floor := 0
	var position int
	hasReachedBasement := false

	for i := 0; i < len(res); i++ {
		if string(res[i]) == "(" {
			floor++
		} else if string(res[i]) == ")" {
			floor--
		}

		if floor < 0 && !hasReachedBasement {
			hasReachedBasement = true
			position = i + 1
		}

	}

	fmt.Println(floor)
	fmt.Println(position)

}

File: 2015/dart/day-01.dart
import 'dart:io';

void main(List<String> args) async {
  var file = File("2015/dart/input.txt");
  var floor = 0;
  var res = await file.readAsBytes();

  res.forEach((element) => String.fromCharCode(element).contains("(")
      ? floor++
      : String.fromCharCode(element).contains(")")
          ? floor--
          : null);

  print(floor);
}
